### ISSUE TITLE
TypeError : Cannot read property 'bind' of undefined

### ISSUE THREAD

**Issue by MathieuDerelle** [webrtcHacks/adapter#939]
# Please read first!
Please use [discuss-webrtc](https://groups.google.com/forum/#!forum/discuss-webrtc) for general technical discussions and questions.

- [x] I have provided steps to reproduce (e.g. a link to a [jsfiddle](https://jsfiddle.net/))
- [x] I have provided browser name, version and adapter.js version
- [x] This issue only happens when adapter.js is used

**Note: If the checkboxes above are not checked (which you do after the issue is posted), the issue will be closed.**

## Versions affected

**Browser name including version (e.g. Chrome 64.0.3282.119)**
Chrome 52.0.2743.98
Chrome 51.0.2704.81
Chrome 50.0.2661.89
Chrome 49.0.2623.112
Chrome 49.0.2623.108
Chrome 49.0.2623.105

Opera 37.0.2192.112031
```
  "userAgent": "Mozilla/5.0 (Linux; Android 6.0.1; CPH1613 Build/MMB29M) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.94 Mobile Safari/537.36 OPR/37.0.2192.112031",
  "userAgentInfo": {
    "browserName": "Opera",
    "browserVersion": "37.0.2192.112031",
    "engineName": "AppleWebKit",
    "engineVersion": "537.36",
    "isBot": false,
    "isMobile": true,
    "os": "Android 6.0.1",
    "platform": "Linux"
  },
```

Android 5.4
```
  "userAgent": "Mozilla/5.0 (Linux; Android 5.0.1; SAMSUNG GT-I9505 Build/LRX22C) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/5.4 Chrome/51.0.2704.106 Mobile Safari/537.36",
  "userAgentInfo": {
    "browserName": "Android",
    "browserVersion": "5.4",
    "engineName": "AppleWebKit",
    "engineVersion": "537.36",
    "isBot": false,
    "isMobile": true,
    "os": "Android 5.0.1",
    "platform": "Linux"
  },
```

**adapter.js (e.g. 6.1.0)**
7.2.1

## Description

TypeError : Cannot read property 'bind' of undefined
```
./node_modules/webrtc-adapter/src/js/chrome/getusermedia.js:175 in bind
./node_modules/webrtc-adapter/src/js/adapter_factory.js:48 in shimGetUserMedia
./node_modules/webrtc-adapter/src/js/adapter_core.js:14
webpack/bootstrap:83 in call
```

## Steps to reproduce

having `navigator.mediaDevices.getUserMedia` null (using one of the device mentionned above)

`import 'webrtc-adapter'`

## Expected results
not raising an error

## Actual results
raising an error

**Comment by fippo** [comment]
adapter@7 doesn't support such old versions anymore -- mainly due to the difficulty of testing.
Do these versions have navigator.mediaDevices but not navigator.mediaDevices.getUserMedia?

**Comment by MathieuDerelle** [comment]
yes, otherwize they would not have passed that check : 
https://github.com/webrtcHacks/adapter/blob/c66c520f2545938a6ca287b1177095c9d95ac9cb/src/js/chrome/getusermedia.js#L16

**Comment by meefik** [comment]
I found a line of the error:
https://github.com/webrtcHacks/adapter/blob/ea47dd2632e15fa77395d1d7d514802d5f0ac71b/src/js/chrome/getusermedia.js#L169-L170

```
Cannot read property 'bind' of undefined
```

Temporary fix:
```javascript
if (!navigator.mediaDevices.getUserMedia && navigator.webkitGetUserMedia) {
  navigator.mediaDevices.getUserMedia = navigator.webkitGetUserMedia;
}
```

**Comment by fippo** [comment]
so digging in the old release notes it seems like M47 shipped navigator.mediaDevices.enumerateDevices while M54 shipped navigator.mediaDevices.getUserMedia. This probably didn't break in the past because adapter had a forward shim for mediaDevices.getUserMedia.

**Comment by fippo** [comment]
Should be fixed in 7.2.2 -- but that still won't polyfill navigator.mediaDevices.getUserMedia on those old versions so YMMV
